### PR TITLE
🐛 Source Square: update following state management changes in the CDK

### PR DESCRIPTION
**Pull request by maxi297**
## What
Breaking changes were done as part of https://github.com/airbytehq/airbyte/commit/a45a1e33413eacccb2b8a83450d1638c48fd2042. This PR fixes the breaking changes for square.

## How
Updating the interfaces that were affected by breaking changes.

## 🚨 User Impact 🚨
None as the code should have the exact same behavior

### CODE REVIEW

**Comment by octavia-squidington-iii** [comment]
##  source-square test report (commit [`6fd5b8f3f2`](https://github.com/airbytehq/airbyte/pull/27762/commits/6fd5b8f3f293280400dedc8f18bec11f941a44fa)) - ❌

⏲️  Total pipeline duration: 54 seconds

| Step                                                                 | Result   |
|:---------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-square/metadata.yaml | ✅        |
| Connector version semver check                                       | ✅        |
| Connector version increment check                                    | ❌        |
| QA checks                                                            | ✅        |
| Code format checks                                                   | ✅        |
| Connector package install                                            | ✅        |
| Build source-square docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                           | ❌        |

🔗 [View the logs here](https://storage.googleapis.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/maxi297-state-changes_square/1687892422/6fd5b8f3f293280400dedc8f18bec11f941a44fa/source-square/1.1.0/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-square test
```

File: airbyte-integrations/connectors/source-square/unit_tests/test_component.py [hunk 1]
```diff
@@ -109,16 +109,15 @@ def test_substream_incremental_sync(state, last_record, expected, expected_strea
...
 
     # Covers the test case for abnormal state that is greater than the current time
     if "updated_at" in state and state["updated_at"] > datetime.now().strftime(DATETIME_FORMAT):
         assert actual_stream_slice["updated_at"] != state["updated_at"]
-        slicer.update_cursor(stream_slice=actual_stream_slice, last_record=last_record)
-        assert slicer.get_stream_state()["updated_at"] < state["updated_at"]
     else:
         assert actual_stream_slice == expected_stream_slice
-        slicer.update_cursor(stream_slice=actual_stream_slice, last_record=last_record)
+        slicer.update_state(stream_slice=actual_stream_slice, last_record=last_record)
```
Comment (maxi297) [inline airbyte-integrations/connectors/source-square/unit_tests/test_component.py:120]: The previous test never actually updated the state using `state` but only using `actual_stream_slice` which didn't have the abnormally high `updated_at=2200-01-01T00:00:00.000000Z`. Therefore, this assertion didn't test anything

**Comment by octavia-squidington-iii** [comment]
##  source-square test report (commit [`1624c39d8e`](https://github.com/airbytehq/airbyte/pull/27762/commits/1624c39d8ee2e5aa1f43e3a4bea4c6446bf51241)) - ❌

⏲️  Total pipeline duration: 1238 seconds

| Step                                                                 | Result   |
|:---------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-square/metadata.yaml | ✅        |
| Connector version semver check                                       | ✅        |
| Connector version increment check                                    | ❌        |
| QA checks                                                            | ✅        |
| Code format checks                                                   | ❌        |
| Connector package install                                            | ✅        |
| Build source-square docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                           | ✅        |
| Acceptance tests                                                     | ✅        |

🔗 [View the logs here](https://storage.googleapis.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/maxi297-state-changes_square/1687895193/1624c39d8ee2e5aa1f43e3a4bea4c6446bf51241/source-square/1.1.0/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-square test
```

**Comment by octavia-squidington-iii** [comment]
##  source-square test report (commit [`0937388616`](https://github.com/airbytehq/airbyte/pull/27762/commits/09373886162eecbda8be86ac532e7be9af273944)) - ✅

⏲️  Total pipeline duration: 1248 seconds

| Step                                                                 | Result   |
|:---------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-square/metadata.yaml | ✅        |
| Connector version semver check                                       | ✅        |
| Connector version increment check                                    | ✅        |
| QA checks                                                            | ✅        |
| Code format checks                                                   | ✅        |
| Connector package install                                            | ✅        |
| Build source-square docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                           | ✅        |
| Acceptance tests                                                     | ✅        |

🔗 [View the logs here](https://storage.googleapis.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/maxi297-state-changes_square/1687896518/09373886162eecbda8be86ac532e7be9af273944/source-square/1.1.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-square test
```

**Review by artem1205 — APPROVED** [review]

**Comment by octavia-squidington-iii** [comment]
##  source-square test report (commit [`bc9f5ff721`](https://github.com/airbytehq/airbyte/pull/27762/commits/bc9f5ff72125b6037fc5d1d296e464cd1eeee651)) - ❌

⏲️  Total pipeline duration: 60 seconds

| Step                                                                 | Result   |
|:---------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-square/metadata.yaml | ✅        |
| Connector version semver check                                       | ✅        |
| Connector version increment check                                    | ✅        |
| QA checks                                                            | ✅        |
| Code format checks                                                   | ✅        |
| Connector package install                                            | ✅        |
| Build source-square docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                           | ❌        |

🔗 [View the logs here](https://storage.googleapis.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/maxi297-state-changes_square/1688051140/bc9f5ff72125b6037fc5d1d296e464cd1eeee651/source-square/1.1.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-square test
```

**Comment by octavia-squidington-iii** [comment]
##  source-square test report (commit [`fe56f515df`](https://github.com/airbytehq/airbyte/pull/27762/commits/fe56f515df8e888a94dd119add0119779cf1560e)) - ✅

⏲️  Total pipeline duration: 21mn43s 

| Step                                                                 | Result   |
|:---------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-square/metadata.yaml | ✅        |
| Connector version semver check                                       | ✅        |
| Connector version increment check                                    | ✅        |
| QA checks                                                            | ✅        |
| Code format checks                                                   | ✅        |
| Connector package install                                            | ✅        |
| Build source-square docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                           | ✅        |
| Acceptance tests                                                     | ✅        |

🔗 [View the logs here](https://storage.googleapis.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/maxi297-state-changes_square/1688062793/fe56f515df8e888a94dd119add0119779cf1560e/source-square/1.1.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-square test
```

**Pull request merged**